omero_marshal/decode/decoders/plate.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Glencoe Software, Inc. All rights reserved.
+#
+# This software is distributed under the terms described by the LICENCE file
+# you can find at the root of the distribution bundle.
+# If the file is missing please request a copy by contacting
+# jason@glencoesoftware.com.
+#
+
+from ... import SCHEMA_VERSION
+from .annotation import AnnotatableDecoder
+from omero.model import PlateI
+from omero.rtypes import rstring
+
+
+class Plate201501Decoder(AnnotatableDecoder):
+
+    TYPE = 'http://www.openmicroscopy.org/Schemas/OME/2015-01#Plate'
+
+    OMERO_CLASS = PlateI
+
+    def decode(self, data):
+        v = super(Plate201501Decoder, self).decode(data)
+        self.set_property(v, 'name', data.get('Name'))
+        self.set_property(v, 'description', data.get('Description'))
+        self.set_property(v, 'columnNamingConvention',
+                          data.get('ColumnNamingConvention'))
+        self.set_property(v, 'rowNamingConvention',
+                          data.get('RowNamingConvention'))
+        self.set_property(v, 'columns', data.get('Columns'))
+        self.set_property(v, 'rows', data.get('Rows'))
+        # self.set_property(v, 'defaultSample', data.get('FieldIndex'))
+        v.setDefaultSample(rstring(data.get('FieldIndex')))
```

Comment:
Pretty sure that @sbesson is on the right track.

One of the rather frustrating things about the base OMERO model objects when doing omero-marshal work is that if you use the wrong `RType` instance when setting up a fixture you basically won't know about it until you start doing the round trip test cases. This is why it's essential to have both encoders and decoders when adding any new type to the infrastructure.

If you're ever in one of these situations and you're struggling to find the particular property that might have been incorrectly populated, print out the OMERO model object and look at each property. If the type is incorrect, Ice will say so. For example:

```
In [64]: a = omero.model.ImageI()

In [65]: a.name = omero.rtypes.rlong(99)

In [66]: a
Out[66]: 
object #0 (::omero::model::Image)
{
    _id = <nil>
    _details = object #1 (::omero::model::Details)
    {
        _owner = <nil>
        _group = <nil>
        _creationEvent = <nil>
        _updateEvent = <nil>
        _permissions = <nil>
        _externalInfo = <nil>
        _call = {}
        _event = <nil>
    }
    _loaded = True
    _version = <nil>
    _series = <nil>
    _acquisitionDate = <nil>
    _archived = <nil>
    _partial = <nil>
    _format = <nil>
    _imagingEnvironment = <nil>
    _objectiveSettings = <nil>
    _instrument = <nil>
    _stageLabel = <nil>
    _experiment = <nil>
    _pixelsSeq = 
    {
    }
    _pixelsLoaded = True
    _wellSamplesSeq = 
    {
    }
    _wellSamplesLoaded = True
    _roisSeq = 
    {
    }
    _roisLoaded = True
    _datasetLinksSeq = 
    {
    }
    _datasetLinksLoaded = True
    _datasetLinksCountPerOwner = {}
    _fileset = <nil>
    _annotationLinksSeq = 
    {
    }
    _annotationLinksLoaded = True
    _annotationLinksCountPerOwner = {}
    _name = <invalid value - expected ::omero::RString>
    _description = <nil>
}
```

Note the `<invalid value - expected ::omero::RString>` for `_name`.